crates/extensions/c8y_mapper_ext/src/operations/mod.rs

```diff
@@ -0,0 +1,215 @@
+//! Utilities for executing Cumulocity operations.
+//!
+//! C8y operations need some special handling by the C8y mapper, which needs to use the C8y HTTP
+//! proxy to report on their progress. Additionally, while executing operations we often need to
+//! send messages to different actors and wait for their results before continuing.
+//!
+//! The operations are always triggered remotely by Cumulocity, and a triggered operation must
+//! always terminate in a success or failure. This status needs to be reported to Cumulocity.
+//!
+//! This module contains:
+//! - data definitions of various states which are necessary to maintain in the mapper
+//! - status and error handing utilities for reporting operation success/failure in different ways
+//!   (MQTT, Smartrest)
```

Comment:
Right, this should rather say just "MQTT", because that's how we actually report the status of the operations, just publishing an MQTT message on the operation channel. The translation to smartrest is handled elsewhere in the mapper.